projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -22,87 +37,149 @@ export class PendingRequestComponent implements OnInit, OnChanges {
+    }
+    this.groupService
+      .getManagedRequests(this.id, sortBy)
+      .subscribe((reqs: PendingRequest[]) => {
+        this.requests = [];
+
+        this.requests = reqs.map((req) => {
+          return {
+            member_id: req.member_id,
+            joining_user: req.joining_user,
+            at: req.at,
+          };
+        });
+      });
+  }
 
-  _setRequestData(reqs: PendingRequest[]) {
-    this.requests = [];
-
-    for (const req of reqs) {
-      this.requests.push({
-        member_id: req.member_id,
-        "joining_user.login": `${req.joining_user.first_name || ""} ${
-          req.joining_user.last_name || ""
-        } (${req.joining_user.login || ""})`,
-        "grade": req.joining_user && req.joining_user.grade ? req.joining_user.grade : null,
-        at: req.at,
+  _handleActionResponse(result: RequestActionResponse, verb: string, msg: string) {
+    if (result.success === true && result.message === "updated") {
+      const succ = _.countBy(result.data, (status: string) => {
+        return ["success", "unchanged"].includes(status);
       });
+
+      if (succ.false === undefined || succ.false === 0) {
+        this.messageService.add({
+          severity: "success",
+          summary: "Success",
+          detail: `${succ.true} request(s) have been ${msg}`,
+          life: TOAST_LENGTH,
+        });
+      } else if (succ.true === undefined || succ.true === 0) {
+        this.messageService.add({
+          severity: "error",
+          summary: "Error",
+          detail: `Unable to ${verb} the selected request(s).`,
+          life: TOAST_LENGTH,
+        });
+      } else {
+        this.messageService.add({
+          severity: "warn",
+          summary: "Partial success",
+          detail: `${succ.true} request(s) have been ${msg}, ${succ.false} could not be executed`,
+          life: TOAST_LENGTH,
+        });
+      }
+
+      this.reloadData();
     }
   }
 
-  constructor(private groupService: GroupService) {}
+  _processRequestError(_err) {
+    this.messageService.add({
+      severity: "error",
+      summary: "Error",
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
+  }
+
+  constructor(
+    private groupService: GroupService,
+    private messageService: MessageService
+  ) {}
 
   ngOnInit() {
     this.panel.push({
       name: "Pending Requests",
       columns: this.columns,
     });
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
+  ngOnChanges(_changes: SimpleChanges) {
+    this.reloadData();
   }
 
-  onExpandWidth(e) {}
+  onExpandWidth(_e) {}
 
-  onClickAccept(e) {
-    this.onAccept.emit(e);
+  onProcessRequest(type: string) {
```

Comment:
1. From my point of view, the input input is not a string as all values but 2 are invalid. I suppose TS supports clean enums.

2. "onProcessRequest" is a bit misleading, "onAcceptOrReject" maybe?

Reply:
fixed.